Sources/ArcGISToolkit/Components/FeatureFormView/AttachmentHelpers/AttachmentCameraController.swift

```diff
@@ -75,3 +75,37 @@ final class CameraControllerCoordinator: NSObject, UIImagePickerControllerDelega
         parent.endCapture()
     }
 }
+
+extension AttachmentCameraController {
+    /// Specifies an action to perform when the camera capture mode has changed from photo to video or vice versa.
+    /// - Parameter action: The new camera capture mode.
+    func onCameraCaptureModeChanged(perform action: @escaping (_: UIImagePickerController.CameraCaptureMode) -> Void) -> Self {
+        self.controller.action = action
+        return self
+    }
+}
+
+/// A wrapper around ``UIImagePickerController``.
+///
+/// Use this wrapper to monitor additional properties like the current camera capture mode (photo/video).
+class AttachmentUIImagePickerController: UIImagePickerController {
+    /// Observes changes to the camera capture mode.
+    var cameraCaptureModeObserver: (any NSObjectProtocol)?
+    
+    /// An action to perform when the camera capture mode changes.
+    var action: ((UIImagePickerController.CameraCaptureMode) -> Void)?
+    
+    override func viewDidAppear(_ animated: Bool) {
+        cameraCaptureModeObserver = NotificationCenter.default.addObserver(forName: NSNotification.Name("SourceFormatDidChange"), object: nil, queue: nil) { _ in
+            Task {
+                await self.action?(self.cameraCaptureMode)
```

Comment:
Always check for warnings. This introduced a warning because the `action` closure is not asynchronous.

Reply:
Hmm, I see it now when using 15.4. I didn't see it because I was using 16.0 while working on this, and 16 doesn't show the error.

Reply:
Promise I wouldn't have merged had I seen it 🙂

Reply:
Remember to always test on the minimum supported Xcode version. Otherwise there's the risk of introducing code that has build warnings/errors when building with a supported version of Xcode.

Reply:
#804